src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -285,11 +284,16 @@ public static Envelope FromTransaction(Transaction transaction)
             header = CreateHeader(eventId);
         }
 
-        var items = new[]
+        var items = new List<EnvelopeItem>
         {
             EnvelopeItem.FromTransaction(transaction)
         };
 
+        if (transaction.TransactionProfiler is { } profiler)
+        {
+            items.Add(EnvelopeItem.FromProfileInfo(profiler.Collect(transaction)));
```

Comment:
If it throws here? And if it doesn't, do we can an empty profile?
Worth commenting the behavior here at least